x/authz/types/generic_authorization.go

```diff
@@ -19,11 +19,19 @@ func NewGenericAuthorization(methodName string) *GenericAuthorization {
 }
 
 // MethodName implements Authorization.MethodName.
-func (cap GenericAuthorization) MethodName() string {
-	return cap.MessageName
+func (authorization GenericAuthorization) MethodName() string {
+	return authorization.MessageName
 }
 
 // Accept implements Authorization.Accept.
-func (cap GenericAuthorization) Accept(msg sdk.ServiceMsg, block tmproto.Header) (updated exported.Authorization, delete bool, err error) {
-	return &cap, false, nil
+func (authorization GenericAuthorization) Accept(ctx sdk.Context, msg sdk.ServiceMsg) (updated exported.Authorization, delete bool, err error) {
+	return &authorization, false, nil
+}
+
+// ValidateBasic implements Authorization.ValidateBasic.
+func (authorization GenericAuthorization) ValidateBasic() error {
+	if !msgservice.IsServiceMsg(authorization.MessageName) {
```

Comment:
I am not sure if counting of "/" in IsServiceMsg is sufficient to check if typeURL satisfies service method name type e.g.   i.e. /cosmos.bank.Msg/Send vs /cosmos.bank.MsgSend